zerver/lib/outgoing_webhook.py

```diff
@@ -119,23 +122,32 @@ def do_rest_call(rest_operation, request_data, event, service_handler, timeout=N
     request_kwargs['timeout'] = timeout
 
     try:
-        response = requests.request(http_method, final_url, data=json.dumps(request_data), **request_kwargs)
+        response = requests.request(http_method, final_url, data=request_data, **request_kwargs)
         if str(response.status_code).startswith('2'):
-            response_data = service_handler.process_success(response, event)
-            succeed_with_message(event, response_data["response_message"])
+            (response_required, response_message) = service_handler.process_success(response, event)
+            if response_required:
+                succeed_with_message(event, response_message)
```

Comment:
Yeah, once you remove the `response_required` return value and have `process_event` just return an `Optional[str]` instead of a tuple, I think you can do this instead:

```
if response_message is not None:
  succeed_with_message(event, response_message)
```